src/php/lib/Grpc/BaseStub.php

```diff
@@ -72,11 +72,21 @@ public function __construct($hostname, $opts, Channel $channel = null)
                                  'ChannelCredentials::create methods');
         }
         if ($channel) {
-            if (!is_a($channel, 'Grpc\Channel')) {
-                throw new \Exception('The channel argument is not a'.
-                                     'Channel object');
+            if (!is_a($channel, 'Grpc\Channel') &&
+                !is_a($channel, 'Grpc\InterceptorChannel')) {
+                throw new \Exception('The channel argument is not a Channel object '.
+                    'or an InterceptorChannel object created by '.
+                    'InterceptorHelper::intercept($channel, Interceptor $interceptor)');
```

Comment:
`Channel::intercept`

Reply:
Sorry I forgot to update Error message. 
It's interceptorChannel::intercept, Channel class is implemented in C code and add a method need to add code in multiple place and not easy to make changes later.

Reply:
How about moving it to `Interceptor` class instead? The point is we would like to hide the entire `InterceptorChannel` from the user.